src/composables/multisig.ts

```diff
@@ -0,0 +1,220 @@
+const MULTISIG_TRANSCATION_EXPARATION_HEIGHT = 50;
+
+const LOCAL_STORAGE_MULTISIG_KEY = `${LOCAL_STORAGE_PREFIX}_multisig`;
+
+function storeMultisig(multisig: IMultisig[]) {
+  window.localStorage.setItem(LOCAL_STORAGE_MULTISIG_KEY, JSON.stringify(multisig));
+}
+
+function getStoredMultisig(): IMultisig[] {
+  const storedMultisig = window.localStorage.getItem(LOCAL_STORAGE_MULTISIG_KEY);
+  return storedMultisig ? JSON.parse(storedMultisig) : [];
+}
+
+const multisig = ref(<IMultisig[]>getStoredMultisig());
+const activeGaAccountId = ref('');
+const isMultisigDashboard = ref(false);
+
+const initPollingWatcher = createPollingBasedOnMountedComponents();
+
+export function useMultisig({ store }: UseMultisigOptions) {
+  const { getSdk } = useSdk({ store });
+
+  const accounts = computed<IAccount[]>(() => store.getters.accounts);
+  const activeNetwork = computed<INetwork>(() => store.getters.activeNetwork);
+
+  async function getTransactionByHash(txHash: string) {
+    return fetchJson(`${activeNetwork.value.multisigBackend}/tx/${txHash}`);
+  }
+
+  function setActiveGaAccountId(gaAccountId: string) {
+    activeGaAccountId.value = gaAccountId;
+  }
+
+  async function updateMultisig() {
+    const sdk = await getSdk();
+
+    const rawMultisigData = ref(<IMultisig[]>[]);
+    try {
+      await Promise.all(accounts.value.map(async ({ address }) => rawMultisigData.value.push(
+        ...(await fetchJson(`${activeNetwork.value.multisigBackend}/${address}`)),
+      )));
+    } catch {
+      // eslint-disable-next-line no-console
+      console.log('failed to fetch multisig');
+    }
+
+    rawMultisigData.value = uniqBy(rawMultisigData.value, 'contractId');
+
+    const result = (await Promise.all(
+      rawMultisigData.value.map(async ({ contractId, ...otherMultisig }) => {
+        const contractInstance = await sdk.getContractInstance({
+          source: multisigContract,
+          contractAddress: contractId,
+        });
+
+        const nonce = (await contractInstance.methods.get_nonce()).decodedResult;
+        const signers = (await contractInstance.methods.get_signers()).decodedResult;
+
+        const consensus = (
+          await contractInstance.methods.get_consensus_info()
+        ).decodedResult as any;
+
+        if (consensus.tx_hash) {
+          consensus.tx_hash = Buffer.from(consensus.tx_hash).toString('hex');
+          // consensus.tx = TxBuilder.unpackTx((await getTransactionByHash(consensus.tx_hash)).tx);
+        }
+        consensus.expiration_height = Number(consensus.expiration_height);
+        consensus.confirmations_required = Number(consensus.confirmations_required);
+
+        return {
+          ...camelcaseKeysDeep(consensus),
+          ...otherMultisig,
+          nonce: Number(nonce),
+          signers,
+          contractId,
+          balance: (await sdk.balance(otherMultisig.gaAccountId)) || 0,
+          address: otherMultisig.gaAccountId,
+          name: i18n.t('multisig.multisig-vault'),
```

Comment:
In this case the avatar would remain the same for every account in `AccountInfo`.